test/distributed/checkpoint/test_state_dict.py

```diff
@@ -895,6 +895,48 @@ def init_model_optim():
 
         self._test_save_load(init_model_optim)
 
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    def test_setting_meta_device_model(self) -> None:
+        # This test verifies that we can set model state dict by a meta device model
+        torch.manual_seed(0)
+        with torch.device("meta"):
+            meta_model = nn.Sequential(*[nn.Linear(4, 4, bias=False) for _ in range(2)])
+            for layer in meta_model:
+                fully_shard(layer)
+            fully_shard(meta_model)
+        with torch.device("cpu"):
+            cpu_model = nn.Sequential(*[nn.Linear(4, 4, bias=False) for _ in range(2)])
+            full_sd = cpu_model.state_dict()
+        ptd_state_dict.set_model_state_dict(
+            meta_model,
+            model_state_dict=full_sd,
+            options=ptd_state_dict.StateDictOptions(full_state_dict=True, strict=False),
+        )
```

Comment:
Can we compare the state after `set_model_state_dict()` with the golden sample (cpu_model in this case)?